llvm/lib/SYCLLowerIR/SanitizerKernelMetadata.cpp

```diff
@@ -38,6 +38,30 @@ PreservedAnalyses SanitizerKernelMetadataPass::run(Module &M,
   auto &DL = M.getDataLayout();
   auto &Ctx = M.getContext();
 
+  // Fix device global type, by wrapping a structure type
+  {
+    assert(KernelMetadata->getValueType()->isArrayTy());
+
+    auto Name = KernelMetadata->getName().str();
+    KernelMetadata->setName(Name + "_del");
+    auto *KernelMetadataOld = KernelMetadata;
+
+    StructType *StructTypeWithArray = StructType::create(Ctx);
+    StructTypeWithArray->setBody(KernelMetadataOld->getValueType());
+
+    KernelMetadata = new GlobalVariable(
+        M, StructTypeWithArray, false, GlobalValue::ExternalLinkage,
+        ConstantStruct::get(StructTypeWithArray,
+                            KernelMetadataOld->getInitializer()),
+        Name, nullptr, GlobalValue::NotThreadLocal, 1);
```

Comment:
Would be good to have an inline comment explaining this magic `1`

Reply:
Done